### PR TITLE
Rspace/core 685/system processes not in history

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
Installs should not be persisted to the history trie

### Does this PR relate to an RChain JIRA issue? 
https://rchain.atlassian.net/browse/CORE-685

### Complete this checklist before you submit the PR
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: rspace/src/main/scala/coop/rchain/rspace/RSpace.scala [hunk 1]
```diff
@@ -92,66 +98,65 @@ class RSpace[C, P, A, R, K](val store: IStore[C, P, A, K], val branch: Branch)(
...
       implicit m: Match[P, A, R]): Option[(K, Seq[R])] = {
     if (channels.length =!= patterns.length) {
       val msg = "channels.length must equal patterns.length"
       logger.error(msg)
       throw new IllegalArgumentException(msg)
     }
-    store.withTxn(store.createTxnWrite()) { txn =>
-      logger.debug(s"""|install: searching for data matching <patterns: $patterns>
+    logger.debug(s"""|install: searching for data matching <patterns: $patterns>
                        |at <channels: $channels>""".stripMargin.replace('\n', ' '))
```
Comment (henrytill) [inline rspace/src/main/scala/coop/rchain/rspace/RSpace.scala]: Alignment off now.

File: rspace/src/main/scala/coop/rchain/rspace/RSpace.scala [hunk 2]
```diff
@@ -92,66 +98,65 @@ class RSpace[C, P, A, R, K](val store: IStore[C, P, A, K], val branch: Branch)(
...
-          Some((continuation, dataCandidates.map(_.datum.a)))
-      }
+        None
+      case Some(_) =>
+        throw new RuntimeException("Installing can be done only on startup")
     }
+
   }
 
+  def install(channels: Seq[C], patterns: Seq[P], continuation: K)(
```
Comment (henrytill) [inline rspace/src/main/scala/coop/rchain/rspace/RSpace.scala]: I'm not necessarily opposed to such a refactor, but if we decide to take this approach, let's break it out into a separate ticket and consistently, apply it to all the verbs.

File: rspace/src/test/scala/coop/rchain/rspace/HistoryActionsTests.scala [hunk 3]
```diff
@@ -438,4 +438,41 @@ trait HistoryActionsTests
   }
 }
 
-class LMDBStoreHistoryActionsTests extends LMDBStoreTestsBase with HistoryActionsTests
+class LMDBStoreHistoryActionsTests extends LMDBStoreTestsBase with HistoryActionsTests {
+  "install" should "not be peristed to the history trie" in withTestSpace { space =>
```
Comment (henrytill) [inline rspace/src/test/scala/coop/rchain/rspace/HistoryActionsTests.scala]: `s/peristed/persisted/`

Comment (henrytill) [inline rspace/src/test/scala/coop/rchain/rspace/HistoryActionsTests.scala]: also `"an install" should`...

File: rspace/src/main/scala/coop/rchain/rspace/RSpace.scala (hunk 2 above)
Comment (lukasz-golebiewski) [reply rspace/src/main/scala/coop/rchain/rspace/RSpace.scala]: I agree, but without this I wouldn't be able to implement the new functionality

**Comment by lukasz-golebiewski** [comment]
Sorry @KentShikama, your comment was displayed to me after I've merged

**Pull request merged**